.github/workflows/uv-lock.yml

```diff
@@ -0,0 +1,21 @@
+# pre-commit.ci does not allow network access, so we run this on github actions
+name: Update uv.lock
+
+on:
+  pull_request:
```

Comment:
The `pull_request` event can only handle the lockfile for PRs from branches within the same repository, not for PRs from branches in a forked repository. To handle the lockfile for PRs from a forked repository, the `pull_request_target` event is needed, and that is the most complicated part. I can get it done, just need some time.